src/umpire/device_allocator_helper.hpp

```diff
@@ -0,0 +1,83 @@
+////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+////////////////////////////////////////////////////////////////////////////
+#ifndef UMPIRE_device_allocator_helper_HPP
+#define UMPIRE_device_allocator_helper_HPP
+
+#include <string.h>
+
+#include "umpire/DeviceAllocator.hpp"
+
+namespace umpire {
+
+/*
+ * Const variable for the limit of unique DeviceAllocator
+ * objects available at once.
+ */
+extern __managed__ int UMPIRE_TOTAL_DEV_ALLOCS;
+
+/*
+ * Global arrays for both host and device which hold the
+ * DeviceAllocator objects created.
+ */
+extern DeviceAllocator* UMPIRE_DEV_ALLOCS_h;
+__device__ extern DeviceAllocator* UMPIRE_DEV_ALLOCS;
+
+/*
+ * Get the DeviceAllocator object specified by either the given
+ * name or id.
+ */
+__host__ __device__ DeviceAllocator getDeviceAllocator(const char* name);
+__host__ __device__ DeviceAllocator getDeviceAllocator(int id);
+
+/*
+ * Check if the DeviceAllocator object specified by the
+ * given id currently exists.
+ */
+__host__ __device__ bool deviceAllocatorExists(int id);
+
+/*
+ * Given a name, find the corresponding ID, if it exists.
+ */
+__host__ __device__ int findDeviceAllocatorID(const char* name);
```

Comment:
I don't think this function needs to be exposed, you can just have the implementation inside the getDeviceAllocator(name) function.